src/Service/VendorService/OverDrive/OverDriveBooksVendorService.php

```diff
@@ -0,0 +1,124 @@
+     *   Api client for the OverDrive API
+     */
+    public function __construct(EventDispatcherInterface $eventDispatcher, EntityManagerInterface $entityManager, LoggerInterface $statsLogger, ClientInterface $httpClient, Client $apiClient)
+    {
+        parent::__construct($eventDispatcher, $entityManager, $statsLogger);
+
+        $this->httpClient = $httpClient;
+        $this->apiClient = $apiClient;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load(): VendorImportResultMessage
+    {
+        if (!$this->acquireLock()) {
+            return VendorImportResultMessage::error(parent::ERROR_RUNNING);
+        }
+
+        $this->loadConfig();
+
+        try {
+            $this->progressStart('Starting eReolen Global import from overdrive API');
+
+            $totalCount = (0 !== $this->limit) ? $this->limit : $this->apiClient->getTotalProducts();
+
+            $batchSize = ($this->limit > 0 && $this->limit < self::BATCH_SIZE) ? $this->limit : self::BATCH_SIZE;
+            $offset = 0;
+
+            do {
+                $products = $this->apiClient->getProducts($batchSize, $offset);
+
+                $isbnImageUrlArray = [];
+                foreach ($products as $product) {
+                    $coverImageUrl = $product->images->cover->href ?? null;
+
+                    foreach ($product->formats as $format) {
+                        foreach ($format->identifiers as $identifier) {
+                            if (IdentifierType::ISBN === strtolower($identifier->type)) {
+                                if (!empty($identifier->value)) {
+                                    $isbnImageUrlArray[$identifier->value] = $coverImageUrl;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                $this->updateOrInsertMaterials($isbnImageUrlArray, IdentifierType::ISBN);
+
+                $this->progressMessageFormatted($this->totalUpdated, $this->totalInserted, $this->totalIsIdentifiers);
+                $this->progressAdvance();
+
+                $offset += self::BATCH_SIZE;
+            } while ($offset < $totalCount);
+
+            $this->logStatistics();
+
+            $this->progressFinish();
+
+            return VendorImportResultMessage::success($this->totalIsIdentifiers, $this->totalUpdated, $this->totalInserted, $this->totalDeleted);
+        } catch (\Exception $exception) {
+            return VendorImportResultMessage::error($exception->getMessage());
+        }
+    }
+
+    /**
+     * Set config from service from DB vendor object.
+     *
+     * @throws UnknownVendorServiceException
+     * @throws IllegalVendorServiceException
+     */
+    private function loadConfig(): void
+    {
+        $libraryAccountEndpoint = $this->getVendor()->getDataServerURI();
+        $clientId = $this->getVendor()->getDataServerUser();
+        $clientSecret = $this->getVendor()->getDataServerPassword();
+
```

Comment:
I known that the client validates them, but should there be a check here if they have not been set?

Reply:
No sure if it adds any value